crates/aptos/src/account/multisig_account.rs

```diff
@@ -116,19 +124,96 @@ impl CliCommand<TransactionSummary> for CreateTransaction {
     }
 
     async fn execute(self) -> CliTypedResult<TransactionSummary> {
-        let payload = MultisigTransactionPayload::EntryFunction(
-            self.entry_function_args.create_entry_function_payload()?,
-        );
-        self.txn_options
-            .submit_transaction(aptos_stdlib::multisig_account_create_transaction(
+        let multisig_transaction_payload_bytes =
+            to_bytes::<MultisigTransactionPayload>(&self.entry_function_args.try_into()?)?;
+        let transaction_payload = if self.store_hash_only {
+            aptos_stdlib::multisig_account_create_transaction_with_hash(
                 self.multisig_account.multisig_address,
-                to_bytes(&payload)?,
-            ))
+                HashValue::sha3_256_of(&multisig_transaction_payload_bytes).to_vec(),
+            )
+        } else {
+            aptos_stdlib::multisig_account_create_transaction(
+                self.multisig_account.multisig_address,
+                multisig_transaction_payload_bytes,
+            )
+        };
+        self.txn_options
+            .submit_transaction(transaction_payload)
             .await
             .map(|inner| inner.into())
     }
 }
 
+/// Check entry function against on-chain transaction proposal payload.
+#[derive(Debug, Parser)]
+pub struct CheckTransaction {
+    #[clap(flatten)]
+    pub(crate) multisig_account_with_sequence_number: MultisigAccountWithSequenceNumber,
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    #[clap(flatten)]
+    pub(crate) entry_function_args: EntryFunctionArguments,
+}
+
+#[async_trait]
+impl CliCommand<serde_json::Value> for CheckTransaction {
+    fn command_name(&self) -> &'static str {
+        "CheckTransactionMultisig"
```

Comment:
@banool I would prefer this schema, but I appended `Multisig` because all of the other multisig commands do as well:

* https://github.com/aptos-labs/aptos-core/blob/a6d6be73ed7460ca6868a5d824375333d04beeeb/crates/aptos/src/account/multisig_account.rs#L85
* https://github.com/aptos-labs/aptos-core/blob/a6d6be73ed7460ca6868a5d824375333d04beeeb/crates/aptos/src/account/multisig_account.rs#L123
* https://github.com/aptos-labs/aptos-core/blob/a6d6be73ed7460ca6868a5d824375333d04beeeb/crates/aptos/src/account/multisig_account.rs#L233
* https://github.com/aptos-labs/aptos-core/blob/a6d6be73ed7460ca6868a5d824375333d04beeeb/crates/aptos/src/account/multisig_account.rs#L265
* https://github.com/aptos-labs/aptos-core/blob/a6d6be73ed7460ca6868a5d824375333d04beeeb/crates/aptos/src/account/multisig_account.rs#L298
* https://github.com/aptos-labs/aptos-core/blob/a6d6be73ed7460ca6868a5d824375333d04beeeb/crates/aptos/src/account/multisig_account.rs#L327

Your thoughts?

Reply:
Ahhh weird, okay thanks, this is fine then.